### ISSUE TITLE
Trying to use React Native Canvas with Gesture Responder in Views

### ISSUE THREAD

**Issue by joshiatharva** [iddan/react-native-canvas#174]
Hi all, 
I'm trying to use React Native Canvas to build a Canvas tool for drawing DFAs. I'm trying 
to create this from the Javascript version (excerpt below).

    var canvas = document.getElementById('canvas');
    var touchX, touchY;
    if (canvas.getContext) {
        var ctx = canvas.getContext('2d');
    }

    function drawCircle(ctx, x, y, size) {
        ctx.beginPath();
        ctx.arc(x, y, size, 0, Math.PI*2, false);
        ctx.stroke();
    }

However, in React Native the ctx variable only exists within the handleCanvas() method in the example given in the repo, whereas I want to use GestureResponder to execute drawCircle() on any touch on the <Canvas />. How can I store/handle the value of ctx to be able to draw outside handleCanvas(), so I can execute drawCircle() on any touch on the view. The RN code is given below: 
  
    handleCanvas = (canvas) => {
      var ctx = canvas.getContext("2d"); 
    }

    drawCircle = (evt) => {
      ctx.beginPath();
      ctx.arc(evt.pageX, evt.pageY, 12, 0, Math.PI*2, true);
      ctx.stroke();
    }

    render() {
      return (
      <View onStartShouldSetResponder={(evt) => true} onResponderRelease= 
         {this.drawCircle}>
         <Canvas ref={this.handleCanvas} />
      </View>
      );
    }

**Comment by iddan** [comment]
You should save `ctx` on `this` to access it from `drawCircle()`

**Comment by joshiatharva** [comment]
Hi @iddan, thanks for the response. Would that be as a variable in constructor, like:
```
constructor() {
    this.ctx = null;
}

handleCanvas = (canvas) = {
    this.ctx = canvas.getContext("2d");
}
```
this.setState({ctx: canvas.getContext("2d")}) always returns undefined so does not work.

**Comment by iddan** [comment]
Yeah, the code you provided is okay. Just make sure to check canvas is not null

**Comment by joshiatharva** [comment]
Hi @iddan, I now have another problem. When running this package with a blank Expo project, I encapsulate the Canvas example with a View in order to style it, but anytime I make some changes to my code and hot-reload the app, I get a RedBox error: "undefined is not an object (at this.handleCanvas)" but the stack trace points to withExpoRoot.js, not any line within my code. Screenshots of the code here:
![image2](https://user-images.githubusercontent.com/44905452/75087620-b3ba0c00-553a-11ea-9db7-693d6756c653.jpeg)
![image1 (1)](https://user-images.githubusercontent.com/44905452/75087621-b6b4fc80-553a-11ea-889a-ecc52d3dd70b.jpeg)
![Video_Moment3](https://user-images.githubusercontent.com/44905452/75087823-fda3f180-553c-11ea-98bb-5f3cccc073c6.jpg)
![Video_Moment](https://user-images.githubusercontent.com/44905452/75087825-0268a580-553d-11ea-938f-ca65dcba1d23.jpg)
![Video_Moment1](https://user-images.githubusercontent.com/44905452/75087827-05fc2c80-553d-11ea-8622-999b399ea9f3.jpg)

**Comment by iddan** [comment]
Please provide code examples with [code blocks](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks). I have not used expo so I'm not sure why is it happening. Maybe someone else here can help?
